admin/src/js/services/session.js

```diff
@@ -28,9 +28,13 @@ const _ = require('lodash/core');
 
       const navigateToLogin = function() {
         $log.warn('User must reauthenticate');
+        const userCtx = getUserCtx();
+        const username = userCtx && userCtx.name;
+        const usernameQueryParam = username ? `username=${username}&` : '';
+        const windowLocationHrefEncoded = encodeURIComponent($window.location.href);
         ipCookie.remove(COOKIE_NAME, { path: '/' });
         userCtxCookieValue = undefined;
-        $window.location.href = `/${Location.dbName}/login?redirect=${encodeURIComponent($window.location.href)}`;
+        $window.location.href = `/${Location.dbName}/login?${usernameQueryParam}redirect=${windowLocationHrefEncoded}`;
```

Comment:
I think it would be much nicer and more readable if we refactor this to use [URLSearchParameters interface](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams): 
```suggestion
        const params = new URLSearchParams();
        params.append('redirect', $window.location.href);
        
        const userCtx = getUserCtx();
        const username = userCtx && userCtx.name;
        if (username) {
          params.append('username', username);
        }
        ipCookie.remove(COOKIE_NAME, { path: '/' });
        userCtxCookieValue = undefined;
        $window.location.href = `/${Location.dbName}/login?${params.toString()}`;
```

Reply:
The same question here:
Do you think `$window.location.href` should be encoded in this way `encodeURIComponent($window.location.href)` as before?